TBS-5987 upgrade to Jena 4.9.0

treimanntq opened the pull request:
- update version number for TQ fork release to 1.1.2-tq-3
- update Jetty version to match Jena
- Jena's Metadata class was removed, so use Version instead
- use Java Objects instead of guava
- implement GraphListener methods from removed GraphListenerBase
- use BiMap directly from guava instead of removed Jena version

treimanntq committed:
TBS-5987 upgrade to Jena 4.9.0

- update version number for TQ fork release to 1.1.2-tq-3
- update Jetty version to match Jena
- Jena's Metadata class was removed, so use Version instead
- use Java Objects instead of guava
- implement GraphListener methods from removed GraphListenerBase
- use BiMap directly from guava instead of removed Jena version

cygri commented on rdf-delta-base/src/main/java/org/seaborne/delta/lib/SystemInfo.java:
```diff
@@ -17,27 +17,19 @@
     public static String systemName()   { return "RDF Delta"; }
-    public static String version()      { return metadata.get(PATH + ".version", "development"); }
-    public static String buildDate()    { return metadata.get(PATH + ".build.datetime", "unknown"); }
+    public static String version()      { return Version.versionForClass(SystemInfo.class).orElse("development"); }
```
Does this work? I think for this to work, the `maven-jar-plugin` needs to be instructed to put `Implementation-Version` into `META-INF/MANIFEST.MF` when building the jar:
```
<plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-jar-plugin</artifactId>
      <configuration>
        <archive>
          <manifest>
            <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
          </manifest>
        </archive>
      </configuration>
    </plugin>
  </plugins>
```
The Jena jars inherit this configuration from the Apache parent jar but I don't see it in rdf-delta so this probably needs to be added to the `rdf-delta` POM.

treimanntq replied:
No, it did not!  Fixed.

cygri commented on rdf-delta-integration-tests/src/test/java/org/seaborne/delta/AbstractTestDeltaLink.java:
```diff
@@ -620,9 +620,20 @@ public void dLink_patch_events_graph_1() {
+            @Override public void notifyDeleteList (Graph g, List<Triple> triples) {}
+            @Override public void notifyDeleteTriple(Graph g, Triple t) {}
+            @Override public void notifyEvent(Graph source, Object value) {}
+            protected void deleteEvent(Triple t) {}
+            protected void addEvent(Triple t) { counter.incrementAndGet(); }
```
Looks like `deleteEvent` can be deleted and `addEvent` should be private

treimanntq replied:
Done

cygri reviewed (commented):
Shouldn't we pull the changes from Andy's `main` before applying our own changes?

A Jena upgrade is a significant change and it seems wrong to do that with only a version number bump from `1.1.2-tq-2` to `1.1.2-tq-3`. I mean essentially we are forking `rdf-delta` here.

treimanntq committed:
Merge branch 'dp-1.1.2-tq-3' into TBS-5987

treimanntq committed:
TBS-5987 PR feedback

- include default implementation entries in manifest
- make version information look like the old version information
- also, make use of Optional the way that Jena does
- remove unneeded method from AbtractTestDeltaLink
- adjust visibility on AbtractTestDeltaLink.addEvent

treimanntq commented:
Updated from `main`.  I used 1.1.2-tq-3 since it is the third update of ours to the latest released version, 1.1.2.

cygri reviewed (approved):
Looks ok now. I would say the version should be 1.2-tq-1, not 1.1.2-tq-3. The latter is appropriate for bugfixes etc but the Jena upgrade is a more significant change. The logic here is that we are anticipating that Andy would bump rdf-delta to 1.2 if he did the Jena upgrade himself. So 1.2-tq-1 is our own tweaked version of that “missing” official 1.2 release.
